// WITH_STDLIB
// !RENDER_ALL_DIAGNOSTICS_FULL_TEXT
// DONT_STOP_ON_FIR_ERRORS
<!CONFLICTING_JVM_DECLARATIONS!>object O {
    <!CONFLICTING_JVM_DECLARATIONS!><!UNSUPPORTED_FEATURE!>context(O)<!>
    @JvmField
    val INSTANCE: O? = null<!>
}<!>
