// DIAGNOSTICS: -UNUSED_PARAMETER
// LANGUAGE: -ProhibitOverloadingBetweenVarargsAndArrays

<!CONFLICTING_JVM_DECLARATIONS!><!CONFLICTING_OVERLOADS_DEPRECATION!>fun foo(vararg x: Int)<!> {}<!>
<!CONFLICTING_JVM_DECLARATIONS!><!CONFLICTING_OVERLOADS_DEPRECATION!>fun foo(x: IntArray)<!> {}<!>

<!CONFLICTING_JVM_DECLARATIONS!>fun foo(vararg x: Int?) {}<!>
<!CONFLICTING_JVM_DECLARATIONS!>fun foo(x: Array<Int>) {}<!>

<!CONFLICTING_JVM_DECLARATIONS!><!CONFLICTING_OVERLOADS_DEPRECATION!>fun foo(vararg nn: Number)<!> {}<!>
<!CONFLICTING_JVM_DECLARATIONS!><!CONFLICTING_OVERLOADS_DEPRECATION!>fun foo(nn: Array<out Number>)<!> {}<!>
