// ISSUE: KTIJ-24099

public class Hierarchy <!SYNTAX!>extends<!> <!SYNTAX!>KotlinOnlyClass<!> <!SYNTAX!><!>{
    <!UNRESOLVED_REFERENCE!>void<!> companionMembers(<!SYNTAX!><!>) {
        <!UNRESOLVED_REFERENCE!>Custom<!> companionlateinitVariable <!SYNTAX!>=<!> KotlinOnlyClass<!SYNTAX!>.<!>companionlateinitVariable<!SYNTAX!><!>;
        <!UNRESOLVED_REFERENCE!>Custom<!> companionlateinitStaticVariable <!SYNTAX!>=<!> KotlinOnlyClass<!SYNTAX!>.<!>companionlateinitStaticVariable<!SYNTAX!><!>;

    }<!SYNTAX!><!>
