// MODULE: m1-common
// FILE: common.kt

<!CONFLICTING_OVERLOADS!>expect <!NO_ACTUAL_FOR_EXPECT!>fun foo()<!><!>
<!CONFLICTING_OVERLOADS!>expect <!NO_ACTUAL_FOR_EXPECT!>fun foo()<!><!>

expect <!NO_ACTUAL_FOR_EXPECT!>fun foo(x: Int)<!>
