// FIR_IDENTICAL
<!NOT_A_MULTIPLATFORM_COMPILATION!>expect<!> fun foo1()
<!NOT_A_MULTIPLATFORM_COMPILATION!>expect<!> val bar1 = <!EXPECTED_PROPERTY_INITIALIZER!>42<!>
<!NOT_A_MULTIPLATFORM_COMPILATION!>expect<!> class Baz1 {
    fun foo()

    class Baz12
}

<!NOT_A_MULTIPLATFORM_COMPILATION!>actual<!> fun foo2() = 42
<!MUST_BE_INITIALIZED!><!NOT_A_MULTIPLATFORM_COMPILATION!>actual<!> val bar2: Int<!>
<!NOT_A_MULTIPLATFORM_COMPILATION!>actual<!> interface Baz2

<!NOT_A_MULTIPLATFORM_COMPILATION!>actual<!> class Baz1 {
    <!NOT_A_MULTIPLATFORM_COMPILATION!>actual<!> fun foo() {
    }
}
