
<!DEPRECATED_MODIFIER, INCOMPATIBLE_MODIFIERS!>header<!> <!INCOMPATIBLE_MODIFIERS!>impl<!> <!NO_ACTUAL_FOR_EXPECT!>class <!ACTUAL_WITHOUT_EXPECT!>First<!><!>

<!DEPRECATED_MODIFIER, INCOMPATIBLE_MODIFIERS!>header<!> <!INCOMPATIBLE_MODIFIERS!>expect<!> <!NO_ACTUAL_FOR_EXPECT!>class Second<!>

<!DEPRECATED_MODIFIER, INCOMPATIBLE_MODIFIERS!>header<!> <!INCOMPATIBLE_MODIFIERS!>actual<!> <!NO_ACTUAL_FOR_EXPECT!>class <!ACTUAL_WITHOUT_EXPECT!>Third<!><!>

<!DEPRECATED_MODIFIER, INCOMPATIBLE_MODIFIERS!>impl<!> <!INCOMPATIBLE_MODIFIERS!>expect<!> <!NO_ACTUAL_FOR_EXPECT!>class <!ACTUAL_WITHOUT_EXPECT!>Fourth<!><!>

<!DEPRECATED_MODIFIER, INCOMPATIBLE_MODIFIERS!>impl<!> <!INCOMPATIBLE_MODIFIERS!>actual<!> class <!ACTUAL_WITHOUT_EXPECT!>Fifth<!>

<!INCOMPATIBLE_MODIFIERS!>expect<!> <!INCOMPATIBLE_MODIFIERS!>actual<!> <!NO_ACTUAL_FOR_EXPECT!>class <!ACTUAL_WITHOUT_EXPECT!>Sixth<!><!>
