// IGNORE_FIR_DIAGNOSTICS
// RUN_PIPELINE_TILL: FRONTEND
// LANGUAGE: +ContextParameters
// MODULE: m1-common
// FILE: common.kt

class A

<!CONTEXT_PARAMETERS_UNSUPPORTED, CONTEXT_PARAMETERS_UNSUPPORTED{JVM}!>context(a: <!DEBUG_INFO_MISSING_UNRESOLVED, DEBUG_INFO_MISSING_UNRESOLVED{JVM}!>A<!>)<!>
expect fun test1()

expect fun test2()

<!CONTEXT_PARAMETERS_UNSUPPORTED, CONTEXT_PARAMETERS_UNSUPPORTED{JVM}!>context(a: <!DEBUG_INFO_MISSING_UNRESOLVED, DEBUG_INFO_MISSING_UNRESOLVED{JVM}!>A<!>)<!>
expect fun test3()

<!CONTEXT_PARAMETERS_UNSUPPORTED, CONTEXT_PARAMETERS_UNSUPPORTED{JVM}!>context(a: <!DEBUG_INFO_MISSING_UNRESOLVED, DEBUG_INFO_MISSING_UNRESOLVED{JVM}!>A<!>)<!>
expect fun test4<!NO_ACTUAL_FOR_EXPECT{JVM}!>()<!>

<!CONTEXT_PARAMETERS_UNSUPPORTED, CONTEXT_PARAMETERS_UNSUPPORTED{JVM}!>context(a: <!DEBUG_INFO_MISSING_UNRESOLVED, DEBUG_INFO_MISSING_UNRESOLVED{JVM}!>A<!>)<!>
expect fun test5<!NO_ACTUAL_FOR_EXPECT{JVM}!>()<!>

expect fun test6<!NO_ACTUAL_FOR_EXPECT{JVM}!>(a: A)<!>

expect fun <!NO_ACTUAL_FOR_EXPECT{JVM}!>A<!>.test7()

// MODULE: m2-jvm()()(m1-common)
// FILE: jvm.kt

<!CONTEXT_PARAMETERS_UNSUPPORTED!>context(a: <!DEBUG_INFO_MISSING_UNRESOLVED!>A<!>)<!>
actual fun test1(){}

<!CONTEXT_PARAMETERS_UNSUPPORTED!>context(a: <!DEBUG_INFO_MISSING_UNRESOLVED!>A<!>)<!>
actual fun test2(){}

actual fun test3(){}

actual fun <!ACTUAL_WITHOUT_EXPECT!>A<!>.test4(){}

actual fun test5<!ACTUAL_WITHOUT_EXPECT!>(a: A)<!>{}

<!CONTEXT_PARAMETERS_UNSUPPORTED!>context(a: <!DEBUG_INFO_MISSING_UNRESOLVED!>A<!>)<!>
actual fun test6<!ACTUAL_WITHOUT_EXPECT!>()<!>{}

<!CONTEXT_PARAMETERS_UNSUPPORTED!>context(a: <!DEBUG_INFO_MISSING_UNRESOLVED!>A<!>)<!>
actual fun test7<!ACTUAL_WITHOUT_EXPECT!>()<!>{}
